libpod/pod.go

```diff
@@ -352,13 +382,13 @@ func (p *Pod) InfraContainerID() (string, error) {
 
 // InfraContainer returns the infra container.
 func (p *Pod) InfraContainer() (*Container, error) {
-	if !p.HasInfraContainer() {
```

Comment:
Why did we remove `HasInfraContainer()`? Seems useful.

Reply:
seemed redundant to me, since if the `id == ""` we return no such ctr.